okf/skills/okf/scripts/render_okf_viewer.py

```
@@ -0,0 +1,295 @@
+    return ref.replace("[[", "").replace("]]", "").split("|")[0].split("#")[0].strip()
+
+
+def collect(root):
+    concepts = []
+    for dp, dn, fn in os.walk(root):
+        dn.sort()
+        for name in sorted(fn):
+            if not name.endswith(".md") or name in RESERVED or name.startswith("_"):
+                continue
+            path = os.path.join(dp, name)
+            fm, body = split_fm(open(path, encoding="utf-8").read())
```

Ruff (0.15.20)

[warning] Use a context manager for opening files

(SIM115)